samples/dotnet/EdgeX509AuthDownstreamDevice/Program.cs

```diff
@@ -299,6 +322,10 @@ static void InstallCACert()
             }
         }
 
+        static X509Certificate2 getTrustedCACertFromFile(string trustedCACertPath) {
```

Comment:
What made you want to move this out? Also if you want to keep this, I would suggest renaming the case of the method to be consistent with others in the file.

Reply:
I discussed this decision with @ancaantochi. We decided that it would be awkward for InstallCACert to return null in the case where there are no certificates to install. Do you think I should have InstallACert() return an X509?

Reply:
More context:

The trusted certificate needs to be visible at a higher scope so that it can be passed into the certificate validation callback. I thought about changing the return type of InstallCACert, but decided against it for the above reason. Instead, GetTrustedCACertFromFile exists and is called in two places. One in the implementation, and one right before it is needed by the callback (this diff has it in the wrong place currently).

Reply:
Also, if we change the return type of InstallCACert(), then we will have to pass the root certificate back into Main() scope and then into GetTransport(). I feel it is better to not do this.

Reply:
I suggest moving the logic to obtain the trusted CA certificate(s) in GetTrustedCACertFromFile and move the error checking etc. in there. Follow that up with refactoring InstallCACert by passing the trusted cert(s) as an arg. Hopefully things should read lot better and be cleaner.

Reply:
This is a good idea. Also, I overlooked the fact that this should work when providing no certificate. The callback needs a X509Certificate object in order to work. Currently brainstorming a way to make this object when not directly specified in env vars.

Reply:
I haven't found an easy way to do this. If the certificate validation callback is necessary for Amqp, and this requires a trusted certificate explicitly defined in order to work, this implies that  IOTEDGE_TRUSTED_CA_CERTIFICATE_PEM_PATH should be a required field in the launchSettings env vars. However, this seems to break the concept that one shouldn't need to know what is in one's trusted certificate store.

 If there was a way to programatically get a trusted certificate path from a hostname then I think this is the way to go.

Reply:
I spoke with @mrohera and we decided it would be best to make the trusted certificate path required